cola/widgets/dag.py

```diff
@@ -111,6 +111,12 @@ def reset_branch_head(self):
     def reset_worktree(self):
         self.with_oid(lambda oid: cmds.do(cmds.ResetWorktree, ref=oid))
 
+    def reset_to_commit(self, reset_type):
+        sha1 = self.selected_sha1()
+        if sha1 is None:
+            return
+        cmds.do(cmds.ResetToCommit, reset_type=reset_type, ref=sha1)
```

Comment:
The constructor requires these two arguments (they don't have default values) so it probably makes sense to reorder these and pass them as `cmds.do(cmds.ResetToCommit, sha1, reset_type)`

Also, the logic is pretty similar to the other "needs an object id" actions here, so it probably makes sense to reuse the `if sha1 is None` logic in the `with_oid()` helper.

```python
self.with_oid(lambda  oid: cmds.do(cmds.ResetToCommit, oid, reset_type))
```